values/src/main/java/org/spine3/time/OffsetDates.java

```diff
@@ -153,28 +153,42 @@ private static OffsetDate changeMonth(OffsetDate offsetDate, int monthDelta) {
         return result;
     }
 
-    private static OffsetDate withDate(OffsetDate offsetDate, LocalDate localDate) {
-        return offsetDate.toBuilder()
-                         .setDate(localDate)
-                         .build();
-    }
-
     /**
      * Obtains offset date changed on specified amount of days.
      *
      * @param offsetDate offset date that will be changed
-     * @param daysDelta a number of days that needs to be added or subtracted that can be either positive or negative
+     * @param daysDelta  a number of days that needs to be added or subtracted that can be either positive or negative
      * @return copy of this offset date with new days value
      */
     private static OffsetDate changeDays(OffsetDate offsetDate, int daysDelta) {
         final OffsetDate result = add(offsetDate, DAY_OF_MONTH, daysDelta);
         return result;
     }
 
+    /**
+     * Returns the same instance of offset date with changed local time parameter.
```

Comment:
It is not the same instance! It's a new one. Read documentation for `withXxx` methods from `java.time`.